### PR TITLE
Civo: update Ubuntu version due to deprecation of 16.04 (bionic)

### PR DESCRIPTION
**Pull request by dirien**
## Description
Update Ubuntu images to 18.04 for the Civo provider

## How Has This Been Tested?

Create Inlets exit node

```bash
go run main.go create -p civo --access-token $CIVO_TOKEN
Using provider: civo
Requesting host: trusting-bose1 in , from civo
2021/07/07 06:02:22 Provisioning host with Civo
e9ac240f-fef1-4aaf-b622-5490dab688b7
Instance ID: c4732871-3b1d-4357-a1df-b6f826c054c7
Host: c4732871-3b1d-4357-a1df-b6f826c054c7, status: 
[1/500] Host: c4732871-3b1d-4357-a1df-b6f826c054c7, status: building
[16/500] Host: c4732871-3b1d-4357-a1df-b6f826c054c7, status: active
inlets PRO TCP (0.8.3) server summary:
  IP: 
  Auth-token: U2NsCrrNVWQ5uSx5d9NhebOPcXYsGZAcMkRboOsov61w1CJDCOQ7qTaGVdp8OHm8
```

## How are existing users impacted? What migration steps/scripts do we need?
none

## Checklist:

I have:

- [x] updated the documentation and/or roadmap (if required)
- [x] read the [CONTRIBUTION](https://github.com/inlets/inlets/blob/master/CONTRIBUTING.md) guide
- [x] signed-off my commits with `git commit -s`
- [ ] added unit tests

### CODE REVIEW

File: cmd/create.go [hunk 1]
```diff
@@ -410,8 +410,8 @@ func createHost(provider, name, region, zone, projectID, userData, inletsPort st
 	} else if provider == "civo" {
 		return &provision.BasicHost{
 			Name:       name,
-			OS:         "811a8dfb-8202-49ad-b1ef-1e6320b20497",
-			Plan:       "g2.small",
+			OS:         "12745392-15c7-4140-925d-441fe7ae57fd",
+			Plan:       "g3.small",
```
Comment (alexellis) [inline cmd/create.go:414]: Did they delete the cheaper g2 small instances? What's the cost of the g3?

File: cmd/create.go [hunk 2]
```diff
@@ -410,8 +410,8 @@ func createHost(provider, name, region, zone, projectID, userData, inletsPort st
 	} else if provider == "civo" {
 		return &provision.BasicHost{
 			Name:       name,
-			OS:         "811a8dfb-8202-49ad-b1ef-1e6320b20497",
-			Plan:       "g2.small",
+			OS:         "12745392-15c7-4140-925d-441fe7ae57fd",
```
Comment (alexellis) [inline cmd/create.go:413]: Can you extract the OS string into a const, just above the return statement? Don't put it at the top of the file.

`const Ubuntu2004 = "12745392-15c7-4140-925d-441fe7ae57fd"`

**Review by alexellis — CHANGES_REQUESTED** [review]
Couple of minor tweaks requested. Thanks

File: cmd/create.go (hunk 1 above)
Comment (dirien) [reply cmd/create.go:414]: image
They dont offer the generation 2 sizes anymore...

image
i was not sure if g3.xsmall is okay from the spec...

Comment (dirien) [reply cmd/create.go:414]: @alexellis what was the spec of the g2.small? 1CPU/1GB or 2GB?

File: cmd/create.go (hunk 2 above)
Comment (dirien) [reply cmd/create.go:413]: Done.

const CivoUbuntu2004ID= ...

File: cmd/create.go (hunk 1 above)
Comment (alexellis) [reply cmd/create.go:416]: From memory it was a 5 USD variant, so let's use the Extra Small variant please 👍

Comment (dirien) [reply cmd/create.go:416]: Okay i found a old list. So i Think g3.small is fine...
image

Comment (dirien) [reply cmd/create.go:416]: ok @alexellis, i change it to g3.xsmall

**Comment by dirien** [comment]
@alexellis: Done

File: cmd/create.go [hunk 3]
```diff
@@ -410,8 +412,8 @@ func createHost(provider, name, region, zone, projectID, userData, inletsPort st
 	} else if provider == "civo" {
 		return &provision.BasicHost{
 			Name:       name,
-			OS:         "811a8dfb-8202-49ad-b1ef-1e6320b20497",
-			Plan:       "g2.small",
+			OS:         CivoUbuntu2004ID,
```
Comment (alexellis) [inline cmd/create.go:415]: Inline the variable and use a lowercase C please. This doesn't need global package scope.

```go
	} else if provider == "civo" {
               const ubuntu2004ID = "d927ad2f-5073-4ed6-b2eb-b8e61aef29a8"
 		return &provision.BasicHost{
```

**Review by alexellis — CHANGES_REQUESTED** [review]
Submitting additional information.

Can you also check whether g3.xsmall is entitled to its own public IP in the Civo plan?

File: cmd/create.go [hunk 4]
```diff
@@ -408,10 +408,11 @@ func createHost(provider, name, region, zone, projectID, userData, inletsPort st
 			Additional: map[string]string{},
 		}, nil
 	} else if provider == "civo" {
+		const civoUbuntu2004ID = "d927ad2f-5073-4ed6-b2eb-b8e61aef29a8"
```
Comment (alexellis) [inline cmd/create.go:411]: `civo` is redundant given the scope, but I am OK with this. Thank you 👍

**Review by alexellis — APPROVED** [review]
Approved for merge

**Comment by alexellis** [comment]
CI is now running, please can you run an e2e test now with the license key that I sent you over Slack DM?

**Comment by alexellis** [comment]
You can bring in the new package now via `go mod`:

https://github.com/inlets/cloud-provision/releases/tag/0.3.1

File: cmd/create.go (hunk 4 above)
Comment (dirien) [reply cmd/create.go:411]: done

**Comment by dirien** [comment]
During e2e tests i found out, that the IP adress for the certifiacte is not the same as the public ip from the server.

this could be due to a different egress router they use... 

image

**Comment by dirien** [comment]
Looks like we run into the know issue https://github.com/inlets/inletsctl/issues/78 thanks @jsiebens  for the hint

**Comment by dirien** [comment]
fix from Civo was implemented.

```bash
➜  ~ inlets-pro tcp client --url "wss://74.220.19.102:8123" \
  --token "8EflECGMZEEmjKny2rh0OQteOoDt7wZp50woVR7kCGmkMS99BtD8NXQaHYFguCyb" \
  --upstream $UPSTREAM \
  --ports $PORTS
2021/07/07 14:38:35 Starting TCP client. Version 0.8.5 - 8db64ded51b1455cf4ad027c52bc9cfbeb55c4b3
2021/07/07 14:38:35 Licensed to: Free trial <contact@openfaas.com>, expires: 30 day(s)
2021/07/07 14:38:35 Upstream server: localhost, for ports: 8080
inlets-pro client. Copyright Alex Ellis, OpenFaaS Ltd 2020
INFO[2021/07/07 14:38:35] Connecting to proxy                           url="wss://74.220.19.102:8123/connect"
INFO[2021/07/07 14:38:35] Connection established.. OK.
```

**Comment by alexellis** [comment]
Approving the CI now. Did you connect a tunnel and test that it worked?

**Comment by dirien** [comment]
Yes, did start a nginx conainer locally:

```bash
➜  ~ curl 74.220.19.102:8080
<!DOCTYPE html>
<html>
<head>
<title>Welcome to nginx!</title>
<style>
    body {
        width: 35em;
        margin: 0 auto;
        font-family: Tahoma, Verdana, Arial, sans-serif;
    }
</style>
</head>
<body>
<h1>Welcome to nginx!</h1>
<p>If you see this page, the nginx web server is successfully installed and
working. Further configuration is required.</p>

<p>For online documentation and support please refer to
<a href="http://nginx.org/">nginx.org</a>.<br/>
Commercial support is available at
<a href="http://nginx.com/">nginx.com</a>.</p>

<p><em>Thank you for using nginx.</em></p>
```

**Comment by alexellis** [comment]
Thank you 👍

**Pull request merged**